Core/clim-basic/geometry/region-utilities.lisp

```diff
@@ -0,0 +1,1204 @@
+
+(defun canonize-polynom (pn)
+  (cond ((= (length pn) 0) pn)
+        ((coordinate= (aref pn 0) 0)
+         (canonize-polynom (subseq pn 1)))
+        (t pn)))
+
+(defun newton-iteration (polynom x-start)
+  ;; ATTENTION: Adapted specifically to our problem, do not use this without
+  ;; reading!
+  (multiple-value-bind (sol done?) (maybe-solve-polynom-trivially polynom)
+    (cond (done?
+           sol)
+          (t
+           (let ((x x-start)
+                 x1
+                 (n 0)
+                 (pn polynom)
+                 (eps-f 0d0)
+                 (eps-f* 0d-16)
+                 (eps-x 1d-20)
+                 (m 20)                 ;maximum number of steps
+                 (res nil))
+             (loop
+               (cond ((> n m)
+                      (return)))
+               (multiple-value-bind (f p2) (horner-schema pn x)
+                 (multiple-value-bind (f*) (horner-schema p2 x)
+                   (cond ((<= (abs f*) eps-f*)
+                          ;; We are stuck at an extremum -- continue with random
+                          ;; starting value
+                          (setf x1 (+ 1d0 (random 2d0))))
+                         (t
+                          (setf x1 (- x (/ f f*)))
+                          (cond ((or (<= (abs f) eps-f)
+                                     (<= (abs (- x1 x)) eps-x))
+                                 ;; a few more steps of newton, to improve
+                                 ;; the result
+                                 (setf x1 (newton-ziel-gerade polynom x1))
+                                 (push x1 res)
+                                 ;; divide (roots)
+                                 (multiple-value-bind (f p2) (horner-schema pn x1)
+                                   f
+                                   (setq pn (canonize-polynom p2))
+                                   (multiple-value-bind (sol done?)
+                                       (maybe-solve-polynom-trivially pn)
+                                     (when done?
+                                       ;; iterate more nonetheless here -- is
+                                       ;; this a good idea?
+                                       (setf sol
+                                             (mapcar (lambda (x)
+                                                       (newton-ziel-gerade
+                                                        polynom x))
+                                                     sol))
+                                       (setf res (nconc sol res))
+                                       (return))))
+                                 (setf x1 x-start)
+                                 (setq n 0))))))
+                 (setf x (min 1d0 (max -1d0 x1)))        ;Is this allowed?
+                 (incf n)))
+             res)))))
+
+(defun horner-schema (polynom x)
+  ;; Evaluates the polynomial `polynom' by means of horner's method at the
```

Comment:
I would call the parameter `polynomial` (`polynom` is German) and change the comment to `;; Evaluate POLYNOMIAL …`.